test/e2e/operator/deployment_api.go

```diff
@@ -97,6 +99,8 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 		Expect(err).ShouldNot(HaveOccurred(), "get events watcher")
 		evCaptured = map[types.UID]map[string]struct{}{}
 		go func() {
+			evWait.Add(1)
+			defer evWait.Done()
```

Comment:
Still slightly broken. The `evWait.Add(1)` has to be outside of the goroutine. 

Otherwise we get:
- goroutine created, but not started yet
- AfterEach is called, `evWait.Wait()` returns immediately, map becomes nil
- goroutine starts to run, processes events, panics